src/useSkeleton.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { useMemo } from 'react'
+import type { SkeletonProps } from './Skeleton'
+import { Skeleton } from './Skeleton'
+
+const emptyProps: SkeletonProps = {}
+
+/**
+ * Returns `withSkeleton` HOC which conditionally returns a loading skeleton in place of contents
+ * */
+export const useSkeleton = (loading: boolean, defaultProps = emptyProps) =>
+    useMemo(
+        () => ({
+            withSkeleton: <T,>(arg: T, props = emptyProps) =>
```

Comment:
I believe this allows passing literally anything for `arg` — we should require it to be a `ReactElement`.

```suggestion
            withSkeleton: (arg: ReactElement, props = emptyProps): ReactElement =>
```

Reply:
Objections to `ReactNode` instead of `ReactElement`?